crates/env_filter/src/parser.rs

```diff
@@ -108,6 +166,11 @@ mod tests {
         assert!(filter.is_none());
     }
 
+    #[test]
+    fn try_parse_spec_invalid_crate() {
+        assert!(try_parse_spec("crate1::mod1=warn=info,crate2=debug", false).is_err());
+    }
```

Comment:
Let's add `snapbox` as a dev dependency and then you can do
```rust
let input = "crate1::mod1=warn=info,crate2=debug";
let actual = try_parse_spec(input).unwrap_err();
assert_data_eq!(actual, str![]);
```
and then use snapshot testing to fill in the expected result.

This will ensure we test the quality of the messages for a given string *and* can easily update the tests if we work to improve the quality of the messages

Reply:
Thanks for the pointers to the `snapbox`. It's really convenient!